rag_experiment_accelerator/evaluation/plain_metrics.py

```diff
@@ -1,153 +1,163 @@
+    results = bleu.compute(predictions=predictions, references=references, max_order=2)
+    # multiplying by 100 to maintain consistency with previous implementation
+    return results["bleu"] * 100
 
 
-def fuzzy_compare_values(value1, value2) -> list[float]:
+def fuzzy(str1: str, str2: str, match_type: str = "token_set_ratio") -> float:
     """
-    Compares two values using fuzzy string matching and appends the similarity score to a list of differences.
+    Compares two strings using fuzzy string matching and returns a similarity score.
 
     Args:
-        value1 (str): The first value to compare.
-        value2 (str): The second value to compare.
+        str1 (str): The first string to compare.
+        str2 (str): The second string to compare.
+        match_type (str): The type of fuzzy string matching to use. Options include:
+            - 'ratio'
+            - 'token_set_ratio'
+            - 'token_sort_ratio'
+            - 'partial_ratio'
+            - 'partial_token_sort_ratio'
+            - 'partial_token_set_ratio'
+            - 'WRatio'
+            - 'QRatio'
 
     Returns:
         A list of the similarity scores.
+
+    Raises:
+        ValueError: If the match type is not recognized.
     """
-    similarity_score = [fuzz.token_set_ratio(value1, value2)]
+    try:
+        fuzzy_match_fn = getattr(fuzz, match_type)
+    except AttributeError:
+        raise ValueError(f"Match type '{match_type}' is not recognized.")
+
+    similarity_score = fuzzy_match_fn(str1, str2)
     return similarity_score
 
 
-def levenshtein(value1, value2):
+def levenshtein(str1: str, str2: str) -> int:
     """
     Calculates the Levenshtein distance between two strings and returns the normalized similarity score as a percentage.
 
     Args:
-        value1 (str): The first string to compare.
-        value2 (str): The second string to compare.
+        str1 (str): The first string to compare.
+        str2 (str): The second string to compare.
 
     Returns:
         int: The normalized similarity score as a percentage.
     """
-    score = int(algorithms.levenshtein.normalized_similarity(value1, value2) * 100)
+    score = d.Levenshtein.normalized_similarity(str1, str2) * 100
     return score
 
 
-def jaccard(value1, value2):
+def jaccard(str1: str, str2: str) -> int:
     """
     Calculates the Jaccard similarity score between two sets of values.
 
     Args:
-        value1 (set): The first set of values.
-        value2 (set): The second set of values.
+        str1 (set): The first set of values.
+        str2 (set): The second set of values.
 
     Returns:
         int: The Jaccard similarity score between the two sets of values, as a percentage.
     """
-    score = int(algorithms.jaccard.normalized_similarity(value1, value2) * 100)
+    score = int(alg.jaccard.normalized_similarity(str1, str2) * 100)
     return score
 
 
-def hamming(value1, value2):
+def hamming(str1: str, str2: str) -> int:
     """
```

Comment:
Severity: Major - Consider adding a check to ensure 'str1' and 'str2' have the same length since the Hamming distance requires equal-length strings. Raise a ValueError if they do not match.


[Hack - peer-review-pal] We are sorry for the intrusion. Feel free to ignore/delete the comments with the tag - they are automatically generated by [our Hack project](https://github.com/microsoft/peer-review-pal).
If you find them useful, want us to delete them or have any feedback or want to contribute - reach out to some of us: juliame@microsoft.com vadimkirilin@microsoft.com oteslenko@microsoft.com vimeena@microsoft.com
You can prevent future comments from appearing by adding #prp-ignore to your PR description.

Reply:
added the pad variable and set to True to account for different length strings. Cant expect the two strings coming in to be of the same length.